homeassistant/helpers/entity_registry.py

```diff
@@ -201,6 +204,14 @@ def async_generate_entity_id(
         Conflicts checked against registered and currently existing entities.
         """
         preferred_string = f"{domain}.{slugify(suggested_object_id)}"
+
+        for param, param_name, max_length in (
+            (domain, "domain", MAX_LENGTH_STATE_DOMAIN),
+            (preferred_string, "preferred_entity_id", MAX_LENGTH_STATE_ENTITY_ID),
+        ):
+            if len(param) > max_length:
+                raise MaxLengthExceeded(param, param_name, max_length)
+
```

Comment:
Since the preferred string can grow by appending numbers its possible we wouldn't catch it.

Maybe check `test_string` at the end against `MAX_LENGTH_STATE_ENTITY_ID`?

Reply:
I had protection in place for that at some point, not sure why I got rid of it. Will go with your idea